src/rabbit.erl

```diff
@@ -442,15 +448,53 @@ environment(App) ->
     lists:keysort(1, [P || P = {K, _} <- application:get_all_env(App),
                            not lists:member(K, Ignore)]).
 
+
 rotate_logs(BinarySuffix) ->
+    rabbit_log:info("Rotating logs with suffix '~s'~n", [BinarySuffix]),
     Suffix = binary_to_list(BinarySuffix),
-    rabbit_log:info("Rotating logs with suffix '~s'~n", [Suffix]),
-    log_rotation_result(rotate_logs(log_location(kernel),
-                                    Suffix,
-                                    rabbit_error_logger_file_h),
-                        rotate_logs(log_location(sasl),
-                                    Suffix,
-                                    rabbit_sasl_report_file_h)).
+    ok = rotate_lager_handlers(Suffix, application:get_env(lager, handlers)),
+    ok = rotate_lager_sinks(Suffix),
+    %% give Lager a moment to perform rotation
+    timer:sleep(2000),
```

Comment:
Isn't there a real way to wait for the actual completion of the rotation instead of using a random timeout?

About the relevance of `rabbitmqctl rotate_logs`, can we change it so it drives lager instead of doing its own rotation? I agree that what RabbitMQ does is limited. Lager must provide a much nicer implementation.